extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/BaseFunction.java

```diff
@@ -61,18 +63,19 @@
         deploymentStatus = error.toString();
     }
 
-    protected HttpResponseMessage dispatch(HttpRequestMessage<Optional<String>> request) {
+    protected HttpResponseMessage dispatch(HttpRequestMessage<Optional<String>> request, ExecutionContext context) {
         try {
-            return nettyDispatch(request);
+            return nettyDispatch(request, context);
         } catch (Exception e) {
             e.printStackTrace();
             return request
                     .createResponseBuilder(HttpStatus.valueOf(500)).build();
         }
     }
 
-    protected HttpResponseMessage nettyDispatch(HttpRequestMessage<Optional<String>> request)
+    protected HttpResponseMessage nettyDispatch(HttpRequestMessage<Optional<String>> request, ExecutionContext context)
             throws Exception {
+        AzureFunctionsLoggerProvider.setAzureFunctionsLogger(context.getLogger());
```

Comment:
If there is a different logger per request, then your solution is not thread safe and I'm not sure it will work if multiple requests are being processed at the same time.

Reply:
Indeed thanks for spotting this. I introduced the threadlocal as you proposed in next comment.